### DISCUSSION TITLE
Need More Explanation on Pagination in the docs

### DISCUSSION THREAD

**Discussion by mohitms007** [TanStack/query#1926] [Q&A]
The docs related to pagination is not very clear like for example what does the results array as a whole contain and how to use it more effectively.

**Comment by TkDodo** [comment]
Are you talking about this page: https://react-query.tanstack.com/guides/paginated-queries

Or about infinite queries?

**Reply by mohitms007** [reply]
Yup I am talking about this link https://react-query.tanstack.com/guides/paginated-queries
Sorry for not including the link.
It doesn't explain very well about the data.hasNext property and paging related details.

**Reply by TkDodo** [reply]
That is probably because paginated queries are just normal queries. The structure depends on what the backend returns. The only thing react-query does when setting keepPreviousData to true is to not go to loading state when you mount a new query key, but it keeps showing the data from the previously mounted query. If there is a `.hasNext`, it’s because the backend returns it.

**Reply by mohitms007** [reply]
Got it thanks for the clarification.
Highly Appreciated!
